x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SSLService.java

```diff
@@ -577,14 +636,17 @@ private void validateServerConfiguration(String prefix) {
      * Due to the nature of KeyStores, this may include certificates that are available, but never used
      * such as a CA certificate that is no longer in use, or a server certificate for an unrelated host.
      *
-     * @see TrustConfig#certificates(Environment)
+     * @see SslTrustConfig#getConfiguredCertificates()
      */
-    public Set<CertificateInfo> getLoadedCertificates() throws GeneralSecurityException, IOException {
-        Set<CertificateInfo> certificates = new HashSet<>();
-        for (SSLConfiguration config : this.getLoadedSSLConfigurations()) {
-            certificates.addAll(config.getDefinedCertificates(env));
-        }
-        return certificates;
+    public Collection<CertificateInfo> getLoadedCertificates() throws GeneralSecurityException, IOException {
+        return this.getLoadedSslConfigurations().stream()
+            .map(SslConfiguration::getConfiguredCertificates)
+            .flatMap(Collection::stream)
+            .map(cert -> new CertificateInfo(
+                cert.getPath(), cert.getFormat(), cert.getAlias(), cert.hasPrivateKey(), cert.getCertificate()
+            ))
+
+            .collect(Sets.toUnmodifiableSortedSet());
```

Comment:
This is needed because `CertificateInfo` implements `ToXContentObject` and `Writeable` which aren't available in ssl-config.

In the old world, `SSLConfiguration.getDefinedCertificates` returned a Collection of `CertificateInfo`
In the new world,  `SslConfiguration.getConfiguredCertificates` returns a  Collection of `StoredCertificate` and X-Pack needs to convert it to `CertificateInfo` (and sorts it, per my comment in `SecurityDocumentationIT`)